File: Java/day6/Day6.java
package day6;

import java.util.concurrent.CopyOnWriteArrayList;

/**
 * Lösung 1 5934
 */
public class Day6 {

    public static void main(String[] args) {
        int[] days = {3, 4, 3, 1, 2};
        // Creating an object of CopyOnWriteArrayList class
        // Declaring object of LanternFish type
        CopyOnWriteArrayList<LanternFish> fishes = new CopyOnWriteArrayList<LanternFish>();

        for (int day : days) {
            fishes.add(new LanternFish(day));
        }

        for (int i = 0; i < 80; i++) {
            for (LanternFish fish : fishes) {
                if (fish.getStartDay() == 0) {
                    fishes.add(new LanternFish(8));
                    fish.minusDay();
                }else {
                    fish.minusDay();
                }
            }
        }

        System.out.printf("Total Fische: %d%n", fishes.size());
    }
}

File: Python/2021/Day 6/Day 6.py
# _*_ coding: utf-8 _*_
'''
Created on

@author: Thomas
'''


# class represents a lanternfish
import os


class Lanternfish():

    def __init__(self, start_day):
        self.start_day = start_day

    def minus_day(self):
        self.start_day -= 1
        if self.start_day < 0:
            self.start_day = 6

    def get_day(self):
        return self.start_day


def load_file():
    ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
    inp = []
    with open(ROOT_DIR + "\input.txt", "r") as file:
        for num in file.read():
            if num.isdigit():
                inp.append(int(num))
    return inp


if __name__ == '__main__':
    days = [3, 4, 3, 1, 2]
    fishes = []
    # days = load_file()
    # create fishes
    for day in days:
        fishes.append(Lanternfish(day))

    # subtract day per fisch
    for i in range(80):
        for fish in fishes:
            if fish.get_day() == 0:
                fishes.append(Lanternfish(9))
                fish.minus_day()
            else:
                fish.minus_day()

    print(f"Total Fische: {len(fishes)}")
